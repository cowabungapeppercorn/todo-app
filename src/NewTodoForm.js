import React from 'react';

class NewTodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      text: "",
      editMode: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.addTodo(this.state);
    this.setState({ text: "" });
  }

  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>

          <label htmlFor="text">Todo:</label>
          <input id="text" name="text"
            value={this.state.text}
            onChange={this.handleChange} />

          <input type="submit" />

        </form>
      </div>
    );
  }
}

export default NewTodoForm; 