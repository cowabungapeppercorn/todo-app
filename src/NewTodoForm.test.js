import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import NewTodoForm from './NewTodoForm';

it('renders without crashing', () => {
  shallow(<NewTodoForm />);
});

it('renders as expected', () => {
  let wrapper = mount(<NewTodoForm />);
  wrapper.setState({
    text: "",
    editMode: false
  });
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});