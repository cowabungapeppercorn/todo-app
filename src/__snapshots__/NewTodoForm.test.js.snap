// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<NewTodoForm>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="text"
      >
        Todo:
      </label>
      <input
        id="text"
        name="text"
        onChange={[Function]}
        value=""
      />
      <input
        type="submit"
      />
    </form>
  </div>
</NewTodoForm>
`;
